torchbenchmark/util/model.py

```diff
@@ -419,3 +422,35 @@ def enable_fx_int8(self, quant_engine:str='x86'):
         except Exception as e:
             print(e)
             raise RuntimeError(f"{self.name} doesn't support `fx_int8` yet!")
+
+    def enable_bf16(self):
+        model_name = self.name
+        try:
+            model, _ = self.get_module()
+            model = model.to(torch.bfloat16)
+        except RuntimeError:
+            warnings.warn(UserWarning(f"{model_name} doesn't support `to(torch.bfloat16)` yet!"))
+            return
+        self.set_module(model)
+        def inputs_convert(example_inputs):
+            if isinstance(example_inputs, torch.Tensor) and example_inputs.dtype == torch.float32:
+                return example_inputs.to(torch.bfloat16)
+            elif isinstance(example_inputs, (tuple, list, dict)):
+                return tree_map(lambda x: inputs_convert(x), example_inputs)
+            else:
+                warnings.warn(UserWarning(f"{model_name} example inputs doesn't convert to `torch.bfloat16`!"))
+                return example_inputs
+        if hasattr(self, 'example_inputs'):
+            self.example_inputs = inputs_convert(self.example_inputs)
+        else:
+            warnings.warn(UserWarning(f"{model_name} example inputs doesn't convert to `torch.bfloat16`!"))
+
+    def enable_amp(self):
+        if not self.dynamo and self.opt_args.backend == 'cudagraph':
+            return NotImplementedError("AMP not implemented for cudagraphs")
+        if self.device == "cuda":
+            self.amp_context = lambda: torch.cuda.amp.autocast(dtype=torch.float16)
```

Comment:
Why don't we just use `lambda: torch.autocast(model.device)` here?

Reply:
`torch.autocast(model.device)` should be the same as `torch.cuda.amp.autocast` or `torch.cpu.amp.autocast`, right? So the difference is `dtype`. Now, `cuda` can support autocast on `float16` and `bfloat16`.

I suppose we didn't test `bfloat16` part in cuda test, and adding `bfloat16` in cuda or adding `float16` in cpu is the trend. So I add the dtype here for future convenience.

Reply:
For `amp`, let's use the default precision set for `torch.cuda.amp.autocast` or `torch.cpu.amp.autocast`, respectively.

Reply:
Done.